gateway/crates/federated-server/src/server.rs

```diff
@@ -61,24 +59,7 @@ pub async fn serve(
     let (sender, mut gateway) = watch::channel(None);
     gateway.mark_unchanged();
 
-    fetch_method
-        .start(
-            GatewayConfig {
-                enable_introspection: config.graph.introspection,
-                operation_limits: config.operation_limits,
-                authentication: config.authentication,
-                subgraphs: config.subgraphs,
-                trusted_documents: config.trusted_documents,
-                wasi: config.hooks,
-                header_rules: config.headers,
-                rate_limit: config.gateway.rate_limit,
-                timeout: config.gateway.timeout,
-                entity_caching: config.entity_caching,
-            },
-            otel_reload,
-            sender,
-        )
-        .await?;
+    fetch_method.start(&config, otel_reload, sender).await?;
```

Comment:
This one here I need to change again, and is a big merge conflict for this PR and mine: the gateway config has the config file location and boolean if hot reloading is enabled on my side. I really also don't want to introduce two new parameters to this function, especially if one of them is boolean.

What I think I'll do is I'll add a new `GatewayConfig` with the config, otel reload, sender, config path and hot reload.